src/provider.py

```diff
@@ -304,3 +315,28 @@ def _on_client_relation_broken(self, event: RelationBrokenEvent) -> None:
 
         # call normal updated handler
         self._on_client_relation_updated(event=event)
+
+    @property
+    def ready(self) -> bool:
```

Comment:
**praise** As before, I believe these properties are very good, both for readability and for having leaner business logic! Well done!